client/src/command.ts

```diff
@@ -0,0 +1,240 @@
+  debug: null,
+  tabSize: null,
+  complementAlias: null,
+  trimBindParam: null,
+  keywordCase: null,
+  identifierCase: null,
+  maxCharPerLine: null,
+  complementOuterKeyword: null,
+  complementColumnAsKeyword: null,
+  removeTableAsKeyword: null,
+  removeRedundantNest: null,
+  complementSqlId: null,
+  convertDoubleColonCast: null,
+  unifyNotEqual: null,
+} satisfies ConfigurationRecord;
+
+type ConfigurationRecord = {
+  configurationFilePath: string;
+  debug: boolean | null | undefined;
+  tabSize: number | null | undefined;
+  complementAlias: boolean | null | undefined;
+  trimBindParam: boolean | null | undefined;
+  keywordCase: string | null | undefined;
+  identifierCase: string | null | undefined;
+  maxCharPerLine: number | null | undefined;
+  complementOuterKeyword: boolean | null | undefined;
+  complementColumnAsKeyword: boolean | null | undefined;
+  removeTableAsKeyword: boolean | null | undefined;
+  removeRedundantNest: boolean | null | undefined;
+  complementSqlId: boolean | null | undefined;
+  convertDoubleColonCast: boolean | null | undefined;
+  unifyNotEqual: boolean | null | undefined;
+};
+
+// WorkspaceConfigurationを受け取り、フォーマッタで利用する設定のみのRecordにして返す
+const extractFormattingConfigurations = (
+  workspaceConfig: WorkspaceConfiguration,
+): Partial<ConfigurationRecord> => {
+  // translate null (that means unsupecified option) to undefined
+  const config: Partial<ConfigurationRecord> = {};
+  for (const key of Object.keys(vsCodeConfigurationsObject)) {
+    const value = workspaceConfig.get(key);
+    if (value != null) {
+      config[key] = value;
+    }
+  }
+  // remove configurationFilePath
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { configurationFilePath, ...restConfiguration } = config;
+
+  return restConfiguration;
+};
+
+const isFileExists = async (uri: Uri): Promise<boolean> => {
+  try {
+    await workspace.fs.stat(uri);
+    return true;
+  } catch (_) {
+    return false;
+  }
+};
+
+// uroborosql-fmt の設定ファイル名を取得する
+// ワークスペースの側で設定されている場合はその値を、設定されていない場合は ".uroborosqlfmtrc.json"を返す
+const getConfigFileName = (
+  defaultName: string = ".uroborosqlfmtrc.json",
+): string => {
+  // uroborosql-fmt の設定を取得
+  const vsCodeConfig = workspace.getConfiguration("uroborosql-fmt");
+
+  // Default value of `uroborosql-fmt.configurationFilePath` is "".
+  const vsCodeConfigPath: string = vsCodeConfig.get("configurationFilePath");
+  return vsCodeConfigPath !== "" ? vsCodeConfigPath : defaultName;
+};
+
+const getTargetFolder = (
+  folders: readonly WorkspaceFolder[],
+): WorkspaceFolder | undefined => {
+  if (folders.length === 1) {
+    return folders[0];
+  } else if (folders.length > 1) {
+    // ワークスペースに複数のフォルダが存在する場合
+    if (!window.activeTextEditor) {
+      // activeTextEditorが存在しない場合
+      return;
+    }
+    const activeEditorPath = window.activeTextEditor.document.uri.fsPath;
+
+    const matchingWorkspace = workspace.workspaceFolders?.find((wsFolder) => {
```

Comment:
（別でコメントした内容を採用した場合これはなくなりますが）
引数で受けた`folders`を使用する方が合理的だと思います。

```suggestion
    const matchingWorkspace = folders.find((wsFolder) => {
```

Reply:
ありがとうございます。ここで `workspaceFolder` を取得しなおす必要はなかったです。
ご提案いただいた関数による共通化を採用したため解消されました 🙏